### PR TITLE
feat: implement ability to append release rules to the default rules set

### PR DESCRIPTION
**Pull request by rustyb**
This PR  comes from a request to not have to copy the full set of default semantic release rules into the input `release-rules` when you want to add one or two additional release rules to the list.

The current behaviour of `release-rules` mean that you override the defaults.

So as not to break any existing implementations i've added a new action input `release-rules-append`. Adding your custom release rules here will allow them to be appended onto the list of existing default rules.

You cannot define both `release-rules` and `release-rules-append` in your actions configuration.

### CODE REVIEW

File: src/utilities/inputProcessors.ts [hunk 1]
```diff
@@ -219,6 +220,26 @@ export const processInputConfigFile = (): string | undefined => {
...
+   * instead of replacing them.
+   */
+
+  if (input.length > 0 && appendInput.length > 0) {
+    throw new Error(
+      'Invalid input release-rules-append and release rules cannot both be used.',
+    );
+  }
+
+  if (appendInput.length > 0 && input.length === 0) {
```
Comment (mjroeleveld) [inline src/utilities/inputProcessors.ts:237]: Nit, `&& input.length === 0` seems redundant.

File: README.md [hunk 2]
```diff
@@ -65,6 +65,20 @@ steps:
...
+      # [
+      #  { "release": "patch", "type": "build" },
+      #  { "release": "patch", "type": "chore" },
+      #  { "release": "patch", "type": "ci" },
+      #  { "release": "patch", "type": "docs" },
+      #  { "release": "patch", "type": "improvement" },
+      #  { "release": "patch", "type": "refactor" },
+      #  { "release": false, "subject": "*\\[skip release\\]*" },
+      # ]
+      # Note: these do not override the default rules. You cannot configure 'release-rules' if you wish to append rules.
```
Comment (mjroeleveld) [inline README.md:79]: Maybe for clarity rephrase to: `cannot be used in conjunction with 'release-rules'` and add to the first sentence.

**Review by mjroeleveld — APPROVED** [review]
Nice work! 🚀

File: src/utilities/inputProcessors.ts (hunk 1 above)
Comment (rustyb) [reply src/utilities/inputProcessors.ts]: Yup I agree 👍

File: README.md (hunk 2 above)
Comment (rustyb) [reply README.md]: Much clearer way to phrase it thanks.

**Pull request merged**